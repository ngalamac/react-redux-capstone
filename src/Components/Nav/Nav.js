import React from 'react';
import { IoIosArrowBack } from 'react-icons/io';
import { useNavigate } from 'react-router-dom';

const Nav = () => {
  const navigate = useNavigate();
  const handleNavigate = () => {
    navigate(-1);
  };
  return (
    <header>
      <nav className="navigation">
        <button
          className="arrow-back  btn-icon"
          type="button"
          onClick={() => handleNavigate()}
        >
          <IoIosArrowBack
            className="icon"
          />
        </button>
        <h3 className="year">2023</h3>
      </nav>
    </header>
  );
};

export default Nav;
