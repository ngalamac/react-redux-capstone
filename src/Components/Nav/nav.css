.navigation {
  display: flex;
  align-items: center;
  background-color: var(--color-secondary-dark);
  padding: 0.5rem 2rem;
  gap: 1rem;
}

.arrow-back {
  display: flex;
  align-items: center;
}

.icon {
  color: var(--color-white);
  font-size: 1.6rem;
  font-weight: bold;
}

.year {
  color: var(--color-white);
}

@media only screen and (min-width: 768px) {
  .navigation {
    height: 4rem;
  }
}
