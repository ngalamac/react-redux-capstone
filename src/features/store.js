import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './countries/countrySlice';

const store = configureStore({
  reducer: {
    country: countryReducer,
  },
});

export default store;
