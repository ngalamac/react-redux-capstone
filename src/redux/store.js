import { configureStore } from '@reduxjs/toolkit';
import teamsReducer from './footballSlice';

const store = configureStore({
  reducer: {
    teams: teamsReducer,
  },
});

export default store;
