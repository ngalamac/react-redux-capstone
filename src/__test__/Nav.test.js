import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Nav from '../Components/Nav/Nav';
import store from '../Redux/configureStore';

describe('The indivial page of the app countires', () => {
  it('renders properly', async () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Nav />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
