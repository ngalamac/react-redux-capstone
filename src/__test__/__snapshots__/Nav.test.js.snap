// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The indivial page of the app countires renders properly 1`] = `
<header>
  <nav
    className="navigation"
  >
    <button
      className="arrow-back  btn-icon"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </button>
    <h3
      className="year"
    >
      2023
    </h3>
  </nav>
</header>
`;
